builtin/set.c

```diff
@@ -135,6 +135,8 @@ static int set(struct mrsh_state *state, int argc, char *argv[],
 		case 'o':
 			if (i + 1 == argc) {
 				print_options(state);
+				state->frame->argc = i + 1;
```

Comment:
We should probably guard this with `if (init_args != NULL)`, otherwise it will also execute when running `set -o` in an already-running shell.

Reply:
No worries! Good to see you around. Is this a good idea:
```patch
@@ -135,7 +135,7 @@ static int set(struct mrsh_state *state, int argc, char *argv[],
                case 'o':
                        if (i + 1 == argc) {
                                print_options(state);
-                               return 0;
+                               goto out; // we must populate state->argv
                        }
                        option = find_long_option(argv[i + 1]);
                        if (!option) {
@@ -199,7 +199,7 @@ static int set(struct mrsh_state *state, int argc, char *argv[],
                argv_free(state->frame->argc, state->frame->argv);
                state->frame->argc = argc - i + 1;
                state->frame->argv = argv_dup(argv_0, state->frame->argc, &argv[i]);
-       } else if (init_args != NULL) {
+       } else out: if (init_args != NULL) {
                // No args given, but we need to initialize state->argv
                state->frame->argc = 1;
                state->frame->argv = argv_dup(strdup(argv[0]), 1, argv);
```

Reply:
Sounds good, except maybe we can put the `out` label before the whole `if` block, it shouldn't affect the logic I think.

Reply:
Alright. Doing that, rebasing and force-pushing.